### PR TITLE
Add tests for lib/common/results.c

### PR DESCRIPTION
**Pull request by waltdisgrace**

### CODE REVIEW

**Comment by kgaillot** [comment]
This is looking good. For the Makefile issue, edit configure.ac and add your new Makefile to AC_CONFIG_FILES().

The autogen.sh script runs automake (via autoreconf), which converts each Makefile.am into a Makefile.in, and autoconf, which generates the configure script from configure.ac. The configure script then converts each Makefile.in to the final Makefile. AC_CONFIG_FILES() tells the configure script which Makefiles need to be created.

File: lib/common/tests/results/pcmk__results.c [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#include <crm_internal.h>
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <glib.h>
+#include <bzlib.h>
+
+static void
+test_for_pcmk_rc_name(void) {
+    g_assert_cmpint(pcmk__strcmp(pcmk_rc_name(pcmk_rc_ok), "pcmk_rc_ok", pcmk__str_none), ==, 0);
```
Comment (kgaillot) [inline lib/common/tests/results/pcmk__results.c:19]: For pcmk_rc_name(), we should test two other cases: one where the value hits the first return, and one where it hits the switch default. (All of the tests should include a case for the default return.)

File: lib/common/tests/results/pcmk__results.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+static void
+test_for_pcmk_rc_str(void) {
+    g_assert_cmpint(pcmk__strcmp(pcmk_rc_str(pcmk_rc_ok), "OK", pcmk__str_none), ==, 0);
+    g_assert_cmpint(pcmk__strcmp(pcmk_rc_str(-1), "Unknown error", pcmk__str_none), ==, 0);
```
Comment (kgaillot) [inline lib/common/tests/results/pcmk__results.c:25]: We should also test the pcmk__rcs[] return. Ideally we'd test the strerror() case as well, but the value there can vary by OS/version, so at most we should check it is non-NULL.

File: lib/common/tests/results/pcmk__results.c [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+static void
+test_for_crm_exit_name(void) {
+    g_assert_cmpint(pcmk__strcmp(crm_exit_name(CRM_EX_OK), "CRM_EX_OK", pcmk__str_none), ==, 0);
+}
+
+static void
+test_for_crm_exit_str(void) {
+    g_assert_cmpint(pcmk__strcmp(crm_exit_str(CRM_EX_OK), "OK", pcmk__str_none), ==, 0);
```
Comment (kgaillot) [inline lib/common/tests/results/pcmk__results.c:35]: We can test both the interrupted and unknown cases as well

**Review by kgaillot — COMMENTED** [review]
Good job putting this together, I think you have a good grasp of how the glib testing works

**Comment by waltdisgrace** [comment]
@kgaillot I finished adding tests, all of which pass on my machine. I couldn't figure out why the Jenkins build is failing just by looking at the Jenkins console output.

**Comment by kgaillot** [comment]
> @kgaillot I finished adding tests, all of which pass on my machine. I couldn't figure out why the Jenkins build is failing just by looking at the Jenkins console output. At least where I was looking, it doesn't seem to state what specific build error there was.

It's the issue in my first comment, the Makefile has to be listed in configure.ac. For Jenkins you have to drill down default -> the # beside pacemaker-build-* -> any configuration -> console output; the relevant part in this case is

    13:32:46 make[5]: Entering directory '/srv/workspace/pacemaker-build-all-voting/pacemaker-build-all-voting/rhel84z-power9-ppc64le/lib/common/tests/results'
    13:32:46 make[5]: *** No rule to make target 'all'.  Stop.

which isn't obvious but indicates the Makefile isn't being built.

**Comment by waltdisgrace** [comment]
@kgaillot Thanks for pointing out where to look. I had forgotten to add configure.ac to my PR. Build passes now!

**Comment by kgaillot** [comment]
Thanks!

**Pull request merged**